### DISCUSSION TITLE
No database entry after email fetch

### DISCUSSION THREAD

**Discussion by chocwithstarfish** [creecros/Mailmagik#38] [Q&A]
Hello, I've been using the Mailmagik plugin with Kanboard for task creation via email. While the plugin successfully fetches emails from my inbox, indicating correct access and email syntax with the receiver formatted as " Project#1<kanboard@abcdef.com> ", it does not create any new tasks. 
I've confirmed this by checking the database directly; no new entries are created, ruling out incorrect project assignment. It seems like everything is configured correctly, yet the emails are processed (and deleted) without task generation. Could you provide insights or solutions for why tasks aren't being created despite successful email retrieval?

**Comment by chocwithstarfish** [comment]
Apologies for any confusion in my previous post. The receiver's email format for task creation is actually with angle brackets, but they were removed in my post (probably for security reasons). 
The email format I use appears to be correct, and Mailmagik is accessing and processing the emails, but still, no tasks are created. I wanted to clarify this to avoid any misunderstanding regarding the email receiver format as a potential cause of the issue

**Comment by creecros** [comment]
Most likely:

1. Your email address that is sending the email, is not authorized within the project.
2. Your syntax is incorrect

**Comment by chocwithstarfish** [comment]
Thank you for the suggestions. Regarding point 1, both the sender's email address and the retrieval account are authorized members of the project. For point 2, since the emails are being fetched and removed from the inbox without creating database entries, is it plausible that the issue is due to incorrect syntax? I've observed that emails without the correct prefix are not fetched by the script, indicating that the syntax for the retrieved emails should be correct. Could there be another reason why tasks aren't being generated?

**Comment by creecros** [comment]
The answer is no, if it is as you say, then the only prevention is user permission within the project.

https://github.com/creecros/Mailmagik/blob/15013ff4d79e6fb8b3a15f7e0d01ab06cc080145/Action/ConvertEmailToTask.php#L138

**Comment by creecros** [comment]
Actually there is 1 more prevention.
But I would consider this syntax. 

https://github.com/creecros/Mailmagik/blob/15013ff4d79e6fb8b3a15f7e0d01ab06cc080145/Action/ConvertEmailToTask.php#L120

So yes, for point 2, your syntax could still be wrong, because it still needs to find a valid project id.

**Comment by creecros** [comment]
How about the Task email function, curious, does that work for you?

**Comment by chocwithstarfish** [comment]
I've attempted to switch from using the TO: format to the SUBJECT format for email task creation by starting the subject line with [Project#1]: followed by the text. Now I have 5 emails in the inbox with this format, and none are being fetched. While this is better than emails being fetched and not added to Kanboard, it still isn't achieving the desired outcome. Both the sender's and receiver's email addresses are set as Project Managers in the project. How is project membership verified? Does it match the sender's name or just the email field in Kanboard? Could there be issues due to special characters like Umlauts in the real name of the sender?

**Reply by creecros** [reply]
It is not enough to simply be a user within kanboard. There is a setting, within each project that allows you to give users permissions. 

https://docs.kanboard.org/v1/user/projects/#project-permissions

**Reply by chocwithstarfish** [reply]
yes, thats done for everyone, every user is even manager instead of "just" user, because I wasn't sure if user is enough.

**Reply by creecros** [reply]
And they show up in your task emails now?

Permissions are verified by email only.

Highly possible special chars could cause issue, try something simple.

**Reply by creecros** [reply]
Is there any chance you know enough to check your error logs?

**Comment by chocwithstarfish** [comment]
![image](https://github.com/creecros/Mailmagik/assets/58484811/8bf0f70d-f2c2-4e9a-ac10-24477bb4366c)

I don't know if this information helps?

**Comment by creecros** [comment]
If they show up in your Task Email:
![image](https://github.com/creecros/Mailmagik/assets/26339368/6386c4d9-9df1-4def-9b9f-647ba54505c0)

And from there, you can Convert to Task, then I'm 99.9% sure your automatic conversion is an issue related to permissions.

As the only difference between this conversion and the automatic conversion, is that there is no user permission check.
